src/test/java/com/rybka/repository/CurrencyHistoryRepositoryTest.java

```diff
@@ -57,17 +55,41 @@ public void testOnCorrectHistoryFindTop5() {
         var historyOtherList = List.of(
                 buildRandomHistory(), buildRandomHistory(), buildRandomHistory()
         );
-        repository.saveAll(historyOtherList);
-        repository.saveAll(historyTop5List);
+
+        historyOtherList
+                .forEach(history -> {
+                    try {
+                        Thread.sleep(1000);
+                        repository.save(history);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                });
+
+        historyTop5List
+                .forEach(history -> {
+                    try {
+                        Thread.sleep(1000);
+                        repository.save(history);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                });
+
+        var expectedTop5HistoryList = repository.findAll();
```

Comment:
You have saved other and expected objects.
The variable name is `expectedTop5HistoryList`. Are you sure that we have exactly 5 objects here? I don't think so,

Reply:
fixed